lib/ansible/template/vars.py

```diff
@@ -30,99 +12,65 @@
 __all__ = ['AnsibleJ2Vars']
 
 
-class AnsibleJ2Vars(Mapping):
-    '''
-    Helper class to template all variable content before jinja2 sees it. This is
-    done by hijacking the variable storage that jinja2 uses, and overriding __contains__
-    and __getitem__ to look like a dict. Added bonus is avoiding duplicating the large
-    hashes that inject tends to be.
+def _process_locals(_l):
+    if _l is None:
+        return {}
+    return {
+        k: v for k, v in _l.items()
+        if v is not missing
+        and k not in ('context', 'environment', 'template')  # NOTE is this really needed?
```

Comment:
```suggestion
        and k not in {'context', 'environment', 'template'}  # NOTE is this really needed?
```

Not really a huge improvement, but a consistent improvement of 1.66µs on a dict with 95 keys.